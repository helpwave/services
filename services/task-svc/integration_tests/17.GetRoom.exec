#!/bin/bash

echo "Exists, has beds:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "b79dc72f-8359-4058-bd7a-e4b7019a5ec3"}' \
	localhost:3001 \
	proto.services.task_svc.v1.RoomService/GetRoom

echo "Exists, has no beds:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "2a3b9111-1f78-4d36-8318-1c69c1e5ad51"}' \
	localhost:3001 \
	proto.services.task_svc.v1.RoomService/GetRoom

echo "Does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "1a3b9111-1f78-4d36-8318-1c69c1e5ad51"}' \
	localhost:3001 \
	proto.services.task_svc.v1.RoomService/GetRoom 2>&1 || echo ""

