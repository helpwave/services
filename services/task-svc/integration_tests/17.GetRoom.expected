Exists, has beds:
{
  "id": "b79dc72f-8359-4058-bd7a-e4b7019a5ec3",
  "name": "Sales",
  "beds": [
    {
      "id": "cb92bad3-68b2-49c1-bfac-494ff18b57f3",
      "name": "Aquamarine"
    },
    {
      "id": "03d95f53-06f1-4163-b9c2-4aa03d1bcc5e",
      "name": "Green"
    },
    {
      "id": "edef8f4f-fa11-471c-9340-c667ae6ecf09",
      "name": "Indigo"
    },
    {
      "id": "31d2cb94-c6fd-4d98-acd4-128c31d51693",
      "name": "Khaki"
    },
    {
      "id": "9a2824d8-cd31-4c5f-82df-1c6bf1d443c6",
      "name": "Pink"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406",
  "wardId": "0f13614c-9043-4af7-a439-cc569ba22639"
}
Exists, has no beds:
{
  "id": "2a3b9111-1f78-4d36-8318-1c69c1e5ad51",
  "name": "Whatever else",
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406",
  "wardId": "0f13614c-9043-4af7-a439-cc569ba22639"
}
Does not exist:
ERROR:
  Code: InvalidArgument
  Message: id not found

